### ISSUE TITLE
TextField updates in debug mode, but doesn't update in release mode

### ISSUE THREAD

**Issue by aprizzo** [flutter/flutter#160426]
### Steps to reproduce

I am running a simple material app with a scaffold and a text field.

### Expected results

I am expecting the text field to work the same way in release and in debug mode

### Actual results

When in debug mode, everything works fine, while in release mode the text field doesn't update when typing the text, it updates only once the keyboard gets closed. No error appears in the log in both cases. 

### Code sample

<details open><summary>Code sample</summary>

```dart
class TestTF extends StatefulWidget {
  const TestTF({super.key});

  @override
  State<TestTF> createState() => _TestTFState();
}

class _TestTFState extends State<TestTF> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(30),
        child: TextField(),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Video demonstration</summary>

Debug mode:

https://github.com/user-attachments/assets/7c5fcab7-00f6-40ec-9ddb-797fef158534

Release mode:

https://github.com/user-attachments/assets/b162e564-f877-47c1-87e5-a9744f5d9e12

</details>

### Logs

<details open><summary>Logs</summary>

```console
No errors appear
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.27.0, on macOS 15.1.1 24B91 darwin-arm64, locale it-IT)
    • Flutter version 3.27.0 on channel stable at /Users/user/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8495dee1fd (7 giorni fa), 2024-12-10 14:23:39 -0800
    • Engine revision 83bacfc525
    • Dart version 3.6.0
    • DevTools version 2.40.2

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • ANDROID_SDK_ROOT = /Users/user/Library/Android/sdk/
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.96.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (6 available)
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 15.1.1 24B91 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 15.1.1 24B91 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 131.0.6778.140

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @aprizzo, what device(s) are you seeing this on? I can't seem to reproduce this. Can you try upgrading to the hotfix (3.27.1) to see if this persists?

**Comment by aprizzo** [comment]
I have tried upgrading to Flutter Channel stable 3.27.1, but the problem is still there. I am trying on a Zebra TC22 with Android 13 and everything works fine, but the very same app shows problems on other Zebra devices, such as the Zebra TC52AX with Android 13 (which is the device from which I recorded the videos I have attached above).

**Comment by danagbemava-nc** [comment]
Hi @aprizzo, did this start after a flutter upgrade? Can we get the full list of affected devices? Can we also get a spec sheet?

**Comment by aprizzo** [comment]
@danagbemava-nc Yes, it started after upgrading to Flutter 3.27.0. As soon as users let me know their devices, I'll let you know the list. As of now I have only received notice that some of them have the problem described above, and I could test it on a single device. Which spec sheet do you need? For the device or for the app?

**Comment by danagbemava-nc** [comment]
> Which spec sheet do you need? For the device or for the app?

For the device. 

Could you also test the master/beta channel to see if you still experience this over there as well?

**Comment by aprizzo** [comment]
I tried running the project on both channels (Master channel v3.28.0-2.0.pre.38540 and Beta channel 3.28.0-0.1.pre) but I couldn't run the project, I was getting errors on most of the packages that my app uses (errors related to io.flutter.plugin.common.PluginRegistry.Registrar) on both channels (both tests have been made using the same Zebra TC52AX with Android 13 mentioned before).

Here are some specs of this device: 
![Screenshot 2024-12-19 alle 14 39 50](https://github.com/user-attachments/assets/bcceb9e6-41ab-4c3e-a36d-025f96ea12cb)
![Screenshot 2024-12-19 alle 14 40 05](https://github.com/user-attachments/assets/47510bc1-70a4-41b5-9156-56513a4d64c7)

If you need other specs, let me know

**Comment by aprizzo** [comment]
The same thing is happening with another device, a Zebra MC3300x with Android 11. Here are its specs: 

![Screenshot 2024-12-19 alle 16 04 46](https://github.com/user-attachments/assets/058a491d-b8c0-42d0-be13-de380219d235)
![Screenshot 2024-12-19 alle 16 05 09](https://github.com/user-attachments/assets/70fb6393-932d-47d0-b6f8-6f69cfbb00e3)

**Comment by danagbemava-nc** [comment]
Thanks for the info. 

Labeling this for further investigation as I cannot reproduce this on my Pixel 7 and I do not have any of the devices on hand to reproduce the issue.

**Comment by aprizzo** [comment]
@danagbemava-nc Sorry for the late reply, I have tried with the counter app on Flutter channel stable, 3.27.1 and it works as it should. What could be the cause of the previous problem? Could it be the environment sdk version that I had set? When I create a new project it's ^3.6.0, while on my own app I have >=3.2.3 <4.0.0

**Comment by danagbemava-nc** [comment]
Hi @aprizzo, that is likely not related. How about in your actual project? Do you still get this issue with stable 3.27.1?

**Comment by aprizzo** [comment]
@danagbemava-nc After having updated the environment sdk version, it seems that the issue has been fixed. Some of the users confirmed me that on those devices they don't see the problem anymore. The app has been built and released through Azure pipelines, which used the Flutter 3.27.1 stable version. I didn't update anything else within my app. So if it's not related to the environment sdk version, it was probably fixed in the Flutter 3.27.1 stable version. It wouldn't still explain why I kept having the problem on my machine, but since it looks to have been fixed, I will close this for now. 
If you manage to find out what was causing the problem or how was it fixed, please let me know so that I can avoid similar issues in the future. 
Thanks for your support.
